program-runtime/src/compute_budget.rs

```diff
@@ -233,6 +274,14 @@ impl ComputeBudget {
         .unwrap_or(MAX_COMPUTE_UNIT_LIMIT)
         .min(MAX_COMPUTE_UNIT_LIMIT) as u64;
 
+        self.accounts_data_size_limit = updated_accounts_data_size_limit
+            .unwrap_or_else(|| {
+                get_default_loaded_accounts_data_limit(&loaded_accounts_data_limit_type)
+            })
+            .min(get_max_loaded_accounts_data_limit(
+                &loaded_accounts_data_limit_type,
+            )) as u64;
```

Comment:
good catch. But zero size doesn't make practical sense, does it? I mean maybe should add check in `compute_budget` to set a floor value for it (`1` maybe?), or fail the ix that sets 0 bytes. Prefer second option to error out at NonZeroUsize conversion.

Reply:
how about this 61e62b70fa27ddd1332880fb2bb213b7e1600a99, leave compute-budget to just process instructions. Call side (eg accounts.rs) is responsible to validate value. `NonZeroUsize` works out perfectly to ensure limit has sensible value, otherwise, we just error out early before loading accounts.

Reply:
Thank you for pointing out the potential panic at first place.

As for `usize` vs `u64`, I actually prefer it be `usize` here in the context of `account.data().len()`. On the `compute_budget` context, it'd be a generic `u64`.